File: oop_chess/src/chess/validators/QueenMoveValidator.java
package chess.validators;

import chess.Board;
import chess.Spot;
import chess.utils.MessageHandler;

public class QueenMoveValidator extends PieceMoveValidator {


    private BishopMoveValidator bishopMoveValidator = new BishopMoveValidator();
    private RookMoveValidator rookMoveValidator = new RookMoveValidator();

    @Override
    public boolean validate(Board board, Spot start, Spot end) {
        return bishopMoveValidator.validate(board, start, end)
                || rookMoveValidator.validate(board, start, end);
    }

    @Override
    protected String getNotValidMessage() {
        return MessageHandler.getHellNoMessage();
    }
}

File: chess_cpp/validators/QueenMoveValidator.cpp
//
// Created by Hrach Khachatryan on 12/15/19.
//

#include "QueenMoveValidator.h"

bool QueenMoveValidator::validate(Board *board, Spot *start, Spot *end) {
    return bishopMoveValidator->validate(board, start, end)
           || rookMoveValidator->validate(board, start, end);}

std::string QueenMoveValidator::getNotValidMessage() {
    return MessageHandler::getHellNoMessage();
}
